CommonClient.py

```diff
@@ -173,10 +174,69 @@ class CommonContext:
     items_handling: typing.Optional[int] = None
     want_slot_data: bool = True  # should slot_data be retrieved via Connect
 
-    # data package
-    # Contents in flux until connection to server is made, to download correct data for this multiworld.
-    item_names: typing.Dict[int, str] = Utils.KeyedDefaultDict(lambda code: f'Unknown item (ID:{code})')
-    location_names: typing.Dict[int, str] = Utils.KeyedDefaultDict(lambda code: f'Unknown location (ID:{code})')
+    class NameLookupDict:
+        """A specialized dict, with helper methods, for id -> name item/location data package lookups by game."""
+        _game_store: typing.Dict[str, typing.ChainMap[int, str]]
```

Comment:
```suggestion
        ctx: CommonContext
        lookup_type: typing.Literal["item", "location"]
        _unknown_item: typing.Callable[[int], str]
        _archipelago_lookup: typing.Dict[int, str]
        _game_store: typing.Dict[str, typing.ChainMap[int, str]]
        _flat_store: typing.Dict[int, str]
```
Some of this was missing type information, and some of it just for completeness.
It's nice to have a place to look with all the members of the class with their type information.

Reply:
Rather than duplicate the definitions in two places, I just moved `_game_store`'s type hint info into `__init__` and made it more explicit for the others.